casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala

```diff
@@ -13,23 +13,20 @@ object PrettyPrinter {
   def buildStringNoLimit(b: Array[Byte]): String = Base16.encode(b)
   def buildStringNoLimit(b: ByteString): String  = Base16.encode(b.toByteArray)
 
-  def buildString(t: GeneratedMessage): String =
+  def buildString(t: CasperMessage): String =
     t match {
       case b: BlockMessage => buildString(b)
-      case d: DeployData   => buildString(d)
       case _               => "Unknown consensus protocol message"
     }
 
+  // TODO shouldn header.parentsHashList be nonempty list?
```

Comment:
please open a ticket for this TODO

Reply:
👍

Reply:
Isn't it empty for genesis? Can we make genesis an exception to this rule somehow?

Reply:
Ticketed https://rchain.atlassian.net/browse/RCHAIN-3846